streams/src/main/java/org/apache/kafka/streams/state/ReadOnlyWindowStore.java

```diff
@@ -0,0 +1,35 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements.  See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License.  You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package org.apache.kafka.streams.state;
+
+import org.apache.kafka.common.annotation.InterfaceStability;
+
+/**
+ * A window store that only supports read operations
+ *
+ * @param <K> Type of keys
+ * @param <V> Type of values
+ */
+@InterfaceStability.Unstable
+public interface ReadOnlyWindowStore<K, V> {
+
+    /**
+     * Get all the key-value pairs with the given key and the time range from all
+     * the existing windows.
+     *
+     * @return an iterator over key-value pairs {@code <timestamp, value>}
+     */
+    WindowStoreIterator<V> fetch(K key, long timeFrom, long timeTo);
```

Comment:
Yeah I think it should be inclusive at both ends, as we stated in the interface:

```
/**
     * Get an iterator over a given range of keys. This iterator MUST be closed after use.
     *
     * @param from The first key that could be in the range
     * @param to The last key that could be in the range
     * @return The iterator for this range.
     * @throws NullPointerException If null is used for from or to.
     */
    KeyValueIterator<K, V> range(K from, K to);
```

And `MemoryNavigableLRUCache.range()` has a bug that needs to be fixed.